### ISSUE TITLE
Traveling System

### ISSUE THREAD

**Issue by TondenOckay** [tinygiant98/darksun-sot#14]
When running a player will use up more of their fatigue points, which they will need to take a short and/or long rest to recover the lost points.

The idea is to have every 3 outdoor areas between cities/towns/villages/points of interest to be equal to 1 days travel when it comes to resting and HTF points.

When zoning/loading from one outdoor area to another outdoor area that isn’t in a city/town/village/points of interest players will lose 30% of their HTF points.

Due to SoT being a PW with all kinds of players playing on it doing different things in different parts of the world we can’t speed up the time in game. But we can have traveling effect things like resing, hunger, thirst, fatigue, and so on.

**Comment by tinygiant98** [comment]
Looking at the possibility of having a setting on each area that affects HTF which will determine a rate of decay of the HTF.  I think that would be a bit more "realistic" that a straight 30% cut just because a PC entered a travel area.  Most of this coding was already accomplished for the other project and is already included in this module, but I'll need to go through it again and ensure it's doing what you intend it to do.

**Comment by TondenOckay** [comment]
Ok that sounds good that should work
What about being able to set a slower movement to an area as well?
For example, you can walk faster on flat grass then you can in deep sand or rocky badlands.
So we could have a setting for areas that you should not be able to move at full speed?

**Comment by tinygiant98** [comment]
Slower than walking?  Yeah, we can do that easily, but it has to be done through NWNX unless you want to do all kinds of messing with 2da, which I don't recommend.  It's just one command with NWNX.

**Comment by TondenOckay** [comment]
That will work, I want to keep things as simple and as clean as possible

**Comment by tinygiant98** [comment]
Yep, it's as simple as adding a single command to the OnAreaEnter and OnAreaExit events for the area, probably via tag-based scripting.

**Comment by TondenOckay** [comment]
Oh and running uses Fatigue points ;)

**Comment by tinygiant98** [comment]
Yup, there's no "running" event, but we might be able to add one, if it's possible in the base game, or capture it via a feat of somet ype.

**Comment by TondenOckay** [comment]
Yeah if we could make it so players always walk unless they turn on run from a feat, I'm all for that.
